proto/spire/agent/svidstore/svidstore.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+package spire.agent.svidstore;
+option go_package = "github.com/spiffe/spire/proto/spire/agent/svidstore";
+
+import "spire/common/common.proto";
+import "spire/common/plugin/plugin.proto";
+import "spire/types/spiffeid.proto";
+
+service SVIDStore {
+    // Puts an X509-SVID in a configured secrets store
+    rpc PutX509SVID(PutX509SVIDRequest) returns (PutX509SVIDResponse);
+
+    // Applies the plugin configuration and returns configuration errors
+    rpc Configure(spire.common.plugin.ConfigureRequest) returns (spire.common.plugin.ConfigureResponse);
+
+    // Returns the version and related metadata of the plugin 
+    rpc GetPluginInfo(spire.common.plugin.GetPluginInfoRequest) returns (spire.common.plugin.GetPluginInfoResponse);
+}
+
+message PutX509SVIDRequest {
+    // X509-SVID to be stored
+    X509SVID svid = 1;
+
+    // Entry selectors with relevant information to store on specific platform
+    repeated spire.common.Selector selectors = 2;
+
+    // Federated bundles to store
+    map<string, bytes> federated_bundles = 3;
+}
+
+message PutX509SVIDResponse {
+}
+
+message X509SVID {
+    // SPIFFE ID of the SVID.
+    spire.types.SPIFFEID  id = 1;
```

Comment:
I chatted with the other maintainers about this, and we ultimately decided we wanted to keep the types used in the SPIRE core RPCs separate from the types used in plugin protocol buffers for maintainability reasons. There are times when we might want to augment API types that may affect the plugin RPCs in undesirable ways if they share the same types, and vice versa. I think after we change this back to a `string`, I am good to merge this.

I apologize for the back and forth on this one since I originally led you down this road. We definitely have some room to improve in protocol buffer management and documentation in the project to make it more clear for contributors to know which types should be used in which RPCs.